sdk/search/search-documents/src/serviceModels.ts

```diff
@@ -2135,5 +2153,117 @@ export interface HnswParameters {
+    /** Polymorphic discriminator, which specifies the different types this object can be */
+    kind: "exhaustiveKnn";
+    /** Contains the parameters specific to exhaustive KNN algorithm. */
+    parameters?: ExhaustiveKnnParameters;
+  };
+
+/** Contains the parameters specific to exhaustive KNN algorithm. */
+export interface ExhaustiveKnnParameters {
+  /** The similarity metric to use for vector comparisons. */
+  metric?: VectorSearchAlgorithmMetric;
+}
+
+/** A dictionary of index projection-specific configuration properties. Each name is the name of a specific property. Each value must be of a primitive type. */
+export interface SearchIndexerIndexProjectionsParameters {
+  /** Describes unknown properties. The value of an unknown property can be of "any" type. */
+  [property: string]: any;
+  /** Defines behavior of the index projections in relation to the rest of the indexer. */
+  projectionMode?: IndexProjectionMode;
+}
+
+/** Definition of additional projections to secondary search indexes. */
+export interface SearchIndexerIndexProjections {
+  /** A list of projections to be performed to secondary search indexes. */
+  selectors: SearchIndexerIndexProjectionSelector[];
+  /** A dictionary of index projection-specific configuration properties. Each name is the name of a specific property. Each value must be of a primitive type. */
+  parameters?: SearchIndexerIndexProjectionsParameters;
+}
+
+/** Contains specific details for a vectorization method to be used during query time. */
+export interface BaseVectorSearchVectorizer {
+  /** Polymorphic discriminator, which specifies the different types this object can be */
+  kind: VectorSearchVectorizerKind;
+  /** The name to associate with this particular vectorization method. */
+  name: string;
+}
+
+/** Contains the parameters specific to using an Azure Open AI service for vectorization at query time. */
+export type AzureOpenAIVectorizer = BaseVectorSearchVectorizer & {
+  /** Polymorphic discriminator, which specifies the different types this object can be */
+  kind: "azureOpenAI";
+  /** Contains the parameters specific to Azure Open AI embedding vectorization. */
+  azureOpenAIParameters?: AzureOpenAIParameters;
+};
+
+/** Contains the parameters specific to generating vector embeddings via a custom endpoint. */
+export type CustomVectorizer = BaseVectorSearchVectorizer & {
+  /** Polymorphic discriminator, which specifies the different types this object can be */
+  kind: "customWebApi";
+  /** Contains the parameters specific to generating vector embeddings via a custom endpoint. */
+  customVectorizerParameters?: CustomVectorizerParameters;
+};
+
+/** Contains the parameters specific to generating vector embeddings via a custom endpoint. */
+export interface CustomVectorizerParameters {
+  /** The uri for the Web API. */
+  uri?: string;
+  /** The headers required to make the http request. */
+  httpHeaders?: { [propertyName: string]: string };
```

Comment:
Can we do `httpHeaders?: Record<string, string>`

Reply:
Done!